crates/bevy_core_pipeline/src/deferred/node.rs

```diff
@@ -0,0 +1,185 @@
+                        load: LoadOp::Load,
+                        store: true,
+                    },
+                }),
+        );
+        if color_attachments.iter().all(Option::is_none) {
+            // All attachments are none: clear the attachment list so that no fragment shader is required.
+            color_attachments.clear();
+        }
+
+        {
+            // Set up the pass descriptor with the depth attachment and optional color attachments.
+            let mut render_pass = render_context.begin_tracked_render_pass(RenderPassDescriptor {
+                label: Some("deferred"),
+                color_attachments: &color_attachments,
+                depth_stencil_attachment: Some(RenderPassDepthStencilAttachment {
+                    view: &view_depth_texture.view,
+                    depth_ops: Some(Operations {
+                        load: if depth_prepass.is_some()
+                            || normal_prepass.is_some()
+                            || motion_vector_prepass.is_some()
+                        {
+                            // If any prepass runs, it will generate a depth buffer so we should use it.
+                            Camera3dDepthLoadOp::Load
+                        } else {
+                            // NOTE: 0.0 is the far plane due to bevy's use of reverse-z projections.
+                            camera_3d.depth_load_op.clone()
+                        }
+                        .into(),
+                        store: true,
+                    }),
+                    stencil_ops: Some(Operations {
+                        load: if depth_prepass.is_some() {
+                            LoadOp::Load // Load if the depth_prepass has run.
+                        } else {
+                            LoadOp::Clear(0)
+                        },
+                        store: true,
+                    }),
+                }),
+            });
+
+            if let Some(viewport) = camera.viewport.as_ref() {
+                render_pass.set_camera_viewport(viewport);
+            }
+
+            // Always run opaque pass to ensure screen is cleared.
+            {
+                // Run the prepass, sorted front-to-back.
+                #[cfg(feature = "trace")]
+                let _opaque_prepass_span = info_span!("opaque_deferred").entered();
+                opaque_deferred_phase.render(&mut render_pass, world, view_entity);
+            }
+
+            if !alpha_mask_deferred_phase.items.is_empty() {
+                // Run the deferred, sorted front-to-back.
+                #[cfg(feature = "trace")]
+                let _alpha_mask_deferred_span = info_span!("alpha_mask_deferred").entered();
+                alpha_mask_deferred_phase.render(&mut render_pass, world, view_entity);
+            }
+        }
+
+        if let Some(prepass_depth_texture) = &view_prepass_textures.depth {
+            // Warnings in WebGL:
```

Comment:
I'm probably going to switch from using a stencil buffer to using a depth buffer for the material id, so hopefully this kind of this wont be an issue.

Reply:
@superdump @JMS55 So it turns out that we also get these warnings in current bevy main when using the prepass in wasm, and this issue is not related to this PR.

Reply:
Made issues:
https://github.com/bevyengine/bevy/issues/9710
https://github.com/bevyengine/bevy/issues/9711

I'll remove the comments from the deferred PR since they are unrelated.